firestore/integration_test_internal/src/android/arena_ref_android_test.cc

```diff
@@ -0,0 +1,175 @@
+
+  ArenaRef arena_ref(env, string);
+  EXPECT_TRUE(arena_ref.get(env).Equals(env, string));
+}
+
+TEST_F(ArenaRefTestAndroid, CopysReferenceToNull) {
+  Env env;
+
+  ArenaRef arena_ref1;
+  ArenaRef arena_ref2(arena_ref1);
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+}
+
+TEST_F(ArenaRefTestAndroid, CopysReferenceToValidObject) {
+  Env env;
+
+  Local<String> string = env.NewStringUtf("hello world");
+
+  ArenaRef arena_ref1(env, string);
+  ArenaRef arena_ref2(arena_ref1);
+
+  EXPECT_TRUE(arena_ref1.get(env).Equals(env, string));
+  EXPECT_TRUE(arena_ref2.get(env).Equals(env, string));
+}
+
+TEST_F(ArenaRefTestAndroid, CopyAssignsReferenceToNull) {
+  Env env;
+
+  ArenaRef arena_ref1, arena_ref2;
+  arena_ref2 = arena_ref1;
+  EXPECT_EQ(arena_ref1.get(env).get(), nullptr);
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+}
+
+TEST_F(ArenaRefTestAndroid, CopyAssignsReferenceToValidObject) {
+  Env env;
+
+  Local<String> string1 = env.NewStringUtf("hello world");
+  Local<String> string2 = env.NewStringUtf("hello earth");
+
+  ArenaRef arena_ref1;
+  ArenaRef arena_ref2(env, string1);
+  ArenaRef arena_ref3(env, string2);
+  arena_ref3 = arena_ref2;
+  arena_ref2 = arena_ref2;
+
+  EXPECT_TRUE(arena_ref3.get(env).Equals(env, string1));
+  EXPECT_TRUE(arena_ref2.get(env).Equals(env, string1));
+
+  arena_ref2 = arena_ref1;
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+  EXPECT_TRUE(arena_ref3.get(env).Equals(env, string1));
+}
+
+TEST_F(ArenaRefTestAndroid, MovesReferenceToNull) {
+  Env env;
+
+  ArenaRef arena_ref1;
+  ArenaRef arena_ref2(std::move(arena_ref1));
+  EXPECT_EQ(arena_ref1.get(env).get(), nullptr);
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+}
+
+TEST_F(ArenaRefTestAndroid, MovesReferenceToValidObject) {
+  Env env;
+
+  Local<String> string = env.NewStringUtf("hello world");
+
+  ArenaRef arena_ref2(env, string);
+  ArenaRef arena_ref3(std::move(arena_ref2));
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+  EXPECT_TRUE(arena_ref3.get(env).Equals(env, string));
+}
+
+TEST_F(ArenaRefTestAndroid, MoveAssignsReferenceToNull) {
+  Env env;
+
+  ArenaRef arena_ref1, arena_ref2;
+  arena_ref2 = std::move(arena_ref1);
+  EXPECT_EQ(arena_ref1.get(env).get(), nullptr);
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+}
+
+TEST_F(ArenaRefTestAndroid, MoveAssignsReferenceToValidObject) {
+  Env env;
+
+  Local<String> string1 = env.NewStringUtf("hello world");
+  Local<String> string2 = env.NewStringUtf("hello earth");
+
+  ArenaRef arena_ref1;
+  ArenaRef arena_ref2(env, string1);
+  arena_ref2 = std::move(arena_ref2);
+  EXPECT_TRUE(arena_ref2.get(env).Equals(env, string1));
+
+  ArenaRef arena_ref3(env, string2);
+  arena_ref3 = std::move(arena_ref2);
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+  EXPECT_TRUE(arena_ref3.get(env).Equals(env, string1));
+
+  arena_ref3 = std::move(arena_ref1);
+  EXPECT_EQ(arena_ref3.get(env).get(), nullptr);
+}
+
+TEST_F(ArenaRefTestAndroid, EnvCallTakeArenaRefTest) {
```

Comment:
Please move this test `EnvCallTakeArenaRefTest` into a new file, `env_android_test.cc`. You obviously don't need to write an exhaustive test suite for the Env class. Just write tests for the new methods that you are adding.